CODEBASE.md

```diff
@@ -4,84 +4,28 @@ This document will give you a high level overview of the modules and components
 
-## Server side rendering
+### In the project root
 
-CC Search runs on our production environments with a Server Side Rendering (SSR) version of the VueJS app. Reading of the [Vue SSR Guide](https://ssr.vuejs.org/guide/) is recommended to understand how it works and the differences that exist compared to running the app on the client-side only.
+The root of the repository contains several configuration files for code formatting, and configuration, for prettier, nuxt, docker, eslint, and other config files
 
-In order to run the SSR app locally, you just need to run the following commands.
+[`nuxt.config.js`](./nuxt.config.js) is an important file. It contains several key pieces of configuration. You can [read more about the NuxtJS config file here](https://nuxtjs.org/guides/configuration-glossary/configuration-build).
 
-```
-$ npm install # if you haven't already
-$ npm run build
-$ npm run server
-```
-
-### Points of attention in the codebase for SSR
-
-When you run `npm run server`, Node will run the [server.js](./server.js) script which starts an [Express.JS](https://expressjs.com/) web server, which will start listening on the configured port and accepting requests and rendering the Vue App, as well as serve static assets.
-
-There are also a few cases of different components for server and client side rendering:
-
-One is the entry file for webpack. You can find them in [clientEntry.js](./src/clientEntry.js) and [serverEntry.js](./src/serverEntry.js). These files are the point of entry of the application, they start the Vue app, the store and the router. There are some minor differences between them, such as polyfill loading and client-side state hydration which are done only on the `clientEntry`.
-
-Because of legacy issues, there are two pages that have separate server and client versions:
-
-- BrowsePage
-- CollectionsBrowsePage
-
-The reason for that is that for some time we used a [JS library](https://www.npmjs.com/package/vue-masonry) to render the image grid. This library used browser specific APIs to render the image grid. Therefore it didn't work on the server renderer. So we had to split those pages into two different versions, one that didn't render the image grid (for the server) and another that did render it (for the client). There is however a [ticket](https://github.com/creativecommons/cccatalog-frontend/issues/934) related to removing this legacy complexity.
-
-For this reason we need to be careful about using client specific APIs (such as `window`, `document`, etc.). If those are necessary, it can only run _only_ on the client side, otherwise it will break on the server and the user will end up seeing an error message when loading the page.
+- environment variables are set in the `export const env = {}` object. Varibles are set with camel case names like this `process.env.varNameGoesHere` and their default values are formatted in the more conventional `process.env.VAR_NAME_GOES_HERE` format.
```

Comment:
Instead of relying on `env`, you might want to use the new [`runtimeConfig`](https://nuxtjs.org/guide/runtime-config/) feature as well.

Reply:
I'd like to, but there's some places in the original codebase where access to environment variables is needed, but `$config` isn't avaliable. After some refactoring we'll switch to `runtimeConfig`.

Reply:
You can set the variables for both, `env` and `publicRuntimeConfig`. Then you can use `process.env.XYZ` whereever needed and migrate the rest to `$config`. Downside of `process.env` is that a rebuild is needed to change the values (while it's not when using `$config`).